src/main/java/gov/nasa/pds/api/engineering/controllers/MyProductsApiBareController.java

```diff
@@ -210,7 +211,10 @@ public String getLatestLidVidFromLid(String lid) throws IOException {
               log.error("Couldn't serialize response for content type " + accept, e);
               return new ResponseEntity<Products>(HttpStatus.INTERNAL_SERVER_ERROR);
           }
-            
+        	catch (ParseCancellationException pce) {
+        		log.error("Could not parse the query string: " + q);
+        		return new ResponseEntity<Products>(HttpStatus.UNPROCESSABLE_ENTITY);
```

Comment:
Hi @al-niessner, sorry I missed to submit these comments earlier

Reply:
> 422 Unprocessable Entity (WebDAV; RFC 4918)
> The request was well-formed but was unable to be followed due to semantic errors.
> 
> This error is a webdav extension. I would not use that also it look like something we want, it is unusual in web APIs.
> 
> Error 400 is what we want.
> 
> Is there a way we can also send a specific error message coming from the parsing. That would be nice to have, unable to parse q parameter:

I am glad that is 422 because it tells me that the problems you were having later are lexer/parser problems and not some other more common error -- kind of how seg_fault masks a host of sins.

I am still going to argue that 422 is correct because from an HTTP standpoint, it is not 400 because the HTTP portion of the game was able to parse the URL and pass all of its parts on to be processed. It was a semantic issue, with respect to the HTTP service not the ANTLR lexer/parser, that caused a failure. Despite my disagreement, I will change it to 400.

Reply:
I added the (%20ref_lid_target%20eq%20urn:nasa:pds:context:target:planet.mercury%20) to the unit test and it did fail. I reduced it to ref_lid_target%20eq%20urn:nasa:pds:context:target:planet.mercury and it still fails. This is good as it is easily reproducible. Will fix.

Check that ref_lid_target is the value you are looking for (urn:nasa:pds:context:target:planet.mercury). If it is not exactly that string, as in is there a version too, it will not find it without a wild card. The 200 with the quoted string says the search was done for that exact string as is and it could not  be found. Hence the question about does it really exist.

Reply:
@al-niessner thanks for your response. 

I found ref_lid_target in properties for a request without criteria. That exists. If it doesn't, this is indeed a good test case, we can keep that aside for now since I don't know if elsticsearch will complain about it, eventually that would be good to have an explicit message saying "variable xxxx does not exists". I believe we would need to go into the registry-dd collection to check that.

Reply:
Ah, I think the quoted text was messed up too. Just updated the unit testing for that as well.

Reply:
All fixed.